### PR TITLE
#161 - Publish Subscribe Channel realisation.

### PR DESCRIPTION
**Pull request by JaXt0r**
Heavily based on iluwatars Message Channel implementation (#143).

### CODE REVIEW

File: publish-subscribe/index.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+layout: pattern
+title: Publish Subscribe
+folder: publish-subscribe
+permalink: /patterns/publish-subscribe/
+categories: Integration
+tags: Java
+---
+
+**Intent:** When applications communicate using a messaging system they do it by using logical addresses
+of the system, so called Publish Subscribe Channel. The publisher broadcasts a message to all registered Subscriber.
```
Comment (mafagafogigante) [inline publish-subscribe/index.md]: You could have pluralized Subscriber. It sounds more natural.

**Comment by iluwatar** [comment]
- [ ] Put under review badge to the pull request
- [ ] Does the example code implement the pattern correctly and follow good coding practices?
- [ ] Does the example code have enough test coverage?
- [ ] Is the example code commented well enough?
- [ ] Is the example code following JavaDoc conventions?
- [ ] Are the project coding conventions being followed?
- [ ] Is the class diagram generated correctly?
- [ ] Is the index.md implemented correctly so the pattern will show correctly on the web site?

**Comment by mafagafogigante** [comment]
I cannot change those checkboxes apparently due to the fact that I am not a contributor here. Anyway, if I was able to do so, my own review would consist of 

- [x] Put under review badge to the pull request
- [ ] Does the example code implement the pattern correctly and follow good coding practices?
- [ ] Does the example code have enough test coverage?
- [x] Is the example code commented well enough?
- [x] Is the example code following JavaDoc conventions?
- [ ] Are the project coding conventions being followed?
- [x] Is the class diagram generated correctly?
- [x] Is the index.md implemented correctly so the pattern will show correctly on the web site?

I must add that besides the grammar detail I pointed out above, I sincerely dislike the idea of having these snippets of code producing command line garbage during testing. Testing should not be just running an example. For instance, it could verify if the subscriber actually gets some stuff the publisher dispatches.

I don't know if anywhere in this project it has been said that the patterns should not pollute the command line during tests, that's why "Are the project coding conventions being followed?" was left unchecked.

Lastly, there is some stuff going on about using Google Java Style, and you are using tabs to indent the code.

**Comment by iluwatar** [comment]
I will work on the issues in another branch.

**Pull request closed without merging**